### ISSUE TITLE
[Bug] Saving emits spurious 'Skipping nested container' warning on freshly-saved .keras files

### ISSUE THREAD

**Issue by rstar327** [keras-team/keras#23021]
### Summary

`keras.saving.load_model` emits multiple `UserWarning: Skipping nested container at '…': no matching group found in the saved file. This usually means the file was saved with a Keras version that did not serialize sublayers nested inside containers. Affected layers will retain their freshly-initialized weights.` even when loading a file that was **just saved with the current version**. The wording suggests weight loss, but in practice the weights round-trip correctly — the warning is a false positive.

The warning was introduced in #22362 to catch legacy files predating nested-container serialization, but it currently also fires for two benign cases:

1. **Containers of pure metadata** — e.g. `variable_serialization_spec`, which is a list of strings (axis/variable names). It holds no `KerasSaveable` and can't have anything to load.
2. **Containers that mirror already-loaded saveables** — e.g. a Functional model's `_operations_by_depth` / `_nodes_by_depth`, whose items are the same layers already loaded via the `layers` collection.

### Reproduction

```python
import os, warnings, tempfile
os.environ["KERAS_BACKEND"] = "tensorflow"
import keras
from keras import layers

inp = layers.Input(shape=(5,))
model = keras.Model(inp, layers.EinsumDense("ab,bc->ac", output_shape=(7,))(inp))
with tempfile.NamedTemporaryFile(suffix=".keras", delete=False) as f:
    path = f.name
model.save(path)
with warnings.catch_warnings(record=True) as w:
    warnings.simplefilter("always")
    keras.models.load_model(path)
print(sum("Skipping nested container" in str(x.message) for x in w))
# Prints 2+ even though the model is freshly saved.
```

Weights actually round-trip correctly — the warning text claiming "Affected layers will retain their freshly-initialized weights" is incorrect for this case.

### Expected

No `Skipping nested container` warning on a freshly-saved file. The warning should still fire for legacy files that genuinely lack a needed nested-container group (covered by `test_legacy_load_skips_missing_nested_containers`).

### Suggested fix

In `keras/src/saving/saving_lib._load_container_state`, before warning, only fire if the in-memory container transitively holds at least one **unvisited** `KerasSaveable`. Pure-metadata containers and containers that mirror already-loaded saveables don't satisfy this and shouldn't warn.

### Impact

Users see scary, repetitive warnings on every `load_model` call with common layers (`EinsumDense`, `Embedding`, `MultiHeadAttention`, every Functional model …) suggesting their weights weren't loaded — when in fact loading worked correctly. Hard to debug, noisy in production, undermines trust.
